otelzap/arrayEncoder.go

```diff
@@ -0,0 +1,111 @@
+package otelzap
+
+import (
+	"fmt"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+type bufferArrayEncoder struct {
```

Comment:
Could you add a brief comment how this works? I assume it implements some interface from zap. If so, we should also add a compile-time check like `var _ zap.SomeInterface = (*bufferArrayEncoder)(nil)`

Reply:
Done. Description and validation are added.